src/app/shared/spaces.service.ts

```diff
@@ -32,22 +32,24 @@ export class SpacesService implements Spaces {
 
   private initRecent(): void {
     // load existing recent spaces from profile.store.recentSpaces
-    this.loadRecent().subscribe((spaces: Space[]) => {
+    this.loadRecent().subscribe((spaces: Space[]): void => {
       this._recent.next(spaces);
     });
 
     // update _recent with newly seen space when spaceChanged is emitted
     this.broadcaster.on<Space>('spaceChanged')
-      .flatMap((changedSpace: Space) => {
-        return this._recent.first().map((recentSpaces: Space[]) => {
-          let index: number = recentSpaces.findIndex((space: Space) => space.id === changedSpace.id);
+      .filter((changedSpace: Space): boolean => {
+        return this._recent.getValue()[0].id !== changedSpace.id;
```

Comment:
Eek, using `getValue` isn't so nice. Is there some other way? Perhaps something like `on('spaceChanged').merge(this._recent.first())`?

Reply:
Sure I can play around with that. Is there any reason in particular getting the value of a BehaviorSubject isn't safe? As far as I can tell BehaviorSubjects only store the last value emitted, so the return from `getValue` should be the current list of spaces.

Reply:
It's just not very idiomatic and doesn't read like Rx. It also makes it significantly more involved to maintain if in the future somebody else comes along and has a reason to change the `BehaviorSubject` to some other kind of `Subject`. IMO declaring the type of `this._recent` as simply `Subject<Space[]>` rather than any *specific* kind of `Subject` would make sense too.

Reply:
That sounds fair to me. I played around with this and had difficulty using merge because the broadcaster emits a space and _recent is `Space[]`, so after merging the type was `Space | Space[]`. 

I did however stumble upon `withLatestFrom`, which looks pretty good for this use case I think. It's similar to `combineLatest`, but instead of emitting whenever either of the observables emits, it only does so for the primary observable.

After the `withLatestFrom`, we have a combination of the changed space and list of recent spaces, which makes it pretty easy for value comparison. I added a `.map()` call prior to the `.filter()` with the purpose of including the (potential) index of the space in recent, so that it could be passed around with the space information and not have to be re-calculated in subsequent functions.

Reply:
Nice, I like that a lot. Looks good to me now.